File: python/wydawanie-reszty/reszta.py
def main():
    N = [200, 100, 50, 20, 10, 5, 2, 1]
    R = int(input("Podaj reszte do wyplacenia: "))
    i = 0
    while R > 0:
        if R >= N[i]:
            P = R // N[i]
            R = R - (N[i] * P)
            print(f"{N[i]} x {P}")
        i += 1

if __name__ == "__main__":
    main()

File: cpp/wydawanie-reszty/reszta.cpp
#include <iostream>
#include <stdlib.h>
 
using namespace std;
 
int main(int argc, char *argv[])
{
//tablica dostepnych nominalow
int N[8]={200, 100, 50, 20, 10, 5, 2, 1};
int R,P, i;
 
cout << "Podaj reszte do wyplacenia: ";
cin >> R;
 
i=0;
while (R>0)       //dopoki nie wydano calej reszty
{
if (R >= N[i])  //sprawdz czy mozna wydac danym nominalem
{
P=R / N[i];   //ile razy wydac dany nominal
R=R-(N[i]*P); //zmniejsz reszte o wydany nominal
cout << N[i] << " x " << P << endl; //wypisz wynik
}
i++;            //rozpatrz kolejny nominal
}
 
//system("PAUSE");
return 0;
}